compiler/semantic/scope.go

```diff
@@ -11,13 +11,19 @@ import (
 
 type Scope struct {
 	parent  *Scope
+	pragmas map[string]any
 	symbols map[string]*entry
 	ctes    map[string]*ast.SQLCTE
 	schema  schema
 }
 
 func NewScope(parent *Scope) *Scope {
-	return &Scope{parent: parent, symbols: make(map[string]*entry), ctes: make(map[string]*ast.SQLCTE)}
+	return &Scope{
+		parent:  parent,
+		symbols: make(map[string]*entry),
+		ctes:    make(map[string]*ast.SQLCTE),
+		pragmas: make(map[string]any),
```

Comment:
Nit: Follow the declaration order.

```suggestion
		pragmas: make(map[string]any),
		symbols: make(map[string]*entry),
		ctes:    make(map[string]*ast.SQLCTE),
```